### ISSUE TITLE
ProtocolVersion: 5 always clears the session

### ISSUE THREAD

**Issue by leshy** [mqttjs/MQTT.js#1139]
Hi,

```js
ClearSession: false
``` 
doesn't seem to have effect in combination with 
```js
ProtocolVersion: 5
```

It seems that this combination of settings always clears the session, server acknowledges this with `sessionPresent: false` in CONACK message

Dropping ProtocolVersion to 4 makes the client behave as expected. Upon some quick investigation I see that CONNECT message flags are calculated to the same value in both cases, so I'm not sure what's the cause of this, but it seems to be on mqtt.js side since I'm seeing expected behaviour using
```sh
mosquitto_sub -d -v -h 127.0.0.1 -p 1883 -t 'testretain' --id test_subscriber --qos 2 --disable-clean-session -V mqttv5
```
vs
```sh
mosquitto_sub -d -v -h 127.0.0.1 -p 1883 -t 'testretain' --id test_subscriber --qos 2 -V mqttv5
```

Thanks!

**Comment by Steve-Mcl** [comment]
Hi @YoDaMa,

I am seeing this exact same issue (`sessionPresent: false` in CONACK message) when using V5 protocol in mqtt.js V4.3.5.  (I realise V4.3.6 is now in NPM but I am not in a position to upgrade just yet)

is this issue still present?

I also confirmed that with `mosquitto_sub ... -V mqttv5` the session is restored (queued messages do arrive) so I know the broker is queing QoS1/2 messages when the session is offline.

Thanks

**Comment by ralight** [comment]
With MQTT v5.0 the flag is technically no longer "clean session", but "clean start", and only affects the clean behaviour when the session is created. The `session expiry interval` property is used to control how long the session should remain active after the client disconnects. In MQTT v3.1.1, the expiry interval is effectively set to infinity for `clean session=false`.

The mosquitto_* tools set the session expiry interval for you when you use `-c` with MQTT v5.0, that's why they appear to be working. I wanted to keep consistent behaviour across protocol versions for these tools. Libraries should be exposing the capability to control clean start/session and the session expiry interval property, but leaving it to the end user to configure, in my opinion.

tldr; I don't think this is a bug.

**Comment by alessandroAmedei** [comment]
@ralight Using library with protocolo Mqtt 5, how do i set the expiryInterval to infinity? 
maybe:
`  properties:{
        sessionExpiryInterval: 4294967295 // 4 bytes integer
      }` ? 

According to the specification [here](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901048)

![image](https://github.com/mqttjs/MQTT.js/assets/28606826/2915097a-75c2-41aa-b972-67b79bea053e)

**Comment by ralight** [comment]
@alessandroAmedei You'd have to check the documentation for the library I'm afraid, I'm not associated with this project and don't know the details.

**Comment by robertsLando** [comment]
@alessandroAmedei Ciao Alessandro,

Yes that's the way, also see docs: https://github.com/mqttjs/MQTT.js#mqttclientstreambuilder-options
